### ISSUE TITLE
Use CMake's FindOpenSSL instead of self-crafted Findopenssl.cmake

### ISSUE THREAD

**Issue by sebastianst** [Corvusoft/restbed#337]
For a long time there's an official [FindOpenSSL](https://cmake.org/cmake/help/v3.6/module/FindOpenSSL.html). Please use that instead of the current home-made version. It doesn't find my system's openssl (Arch Linux here).

**Comment by sebastianst** [comment]
Arch doesn't ship the static libs. However, static libs should only be required if static building is enabled anyways. `FindOpenSSL` allows for checking and setting static biulding.

**Comment by sebastianst** [comment]
Also, I just build an application that statically links restbed in an Ubuntu docker container. Linking fails with
```
/usr/bin/x86_64-linux-gnu-ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcrypto.a(dso_dlfcn.o): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
The reason is that in this setting, linking has to add `-ldl`, which can be achieved by changing the `restbed-static` target_link_libraries call in `CMakeLists.txt` to

```
if ( BUILD_SSL )
        target_link_libraries( ${STATIC_LIBRARY_NAME} LINK_PRIVATE ${ssl_LIBRARY_STATIC} ${crypto_LIBRARY_STATIC} ${CMAKE_DL_LIBS})
```
that is, append ` ${CMAKE_DL_LIBS}`. Please add this quick fix.
Better even would be to use cmake's OpenSSL::SSL link target, which would take care of this.

**Comment by ben-crowhurst** [comment]
The looks like a great way for us to remove some configuration/code.  Will schedule for 4.7 release , thanks.

**Comment by ben-crowhurst** [comment]
Note: We will still be defaulting to the OpenSSL version within the `restbed/dependency` folder and look for other installations if not found.

**Comment by ben-crowhurst** [comment]
This issue has been resolved in 4.7 release branch.
